File: backend/src/NJS.Domain/Entities/Feature.cs
using System.ComponentModel.DataAnnotations;

namespace NJS.Domain.Entities
{
    public class Feature
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }
        
        public bool IsActive { get; set; } = true;

        public virtual ICollection<SubscriptionPlanFeature> SubscriptionPlanFeatures { get; set; }
    }
}

File: frontend/src/types/Feature.ts
export interface Feature {
  id: number;
  name: string;
  description: string;
  isActive: boolean;
}

export interface CreateFeatureRequest {
  name: string;
  description: string;
  isActive: boolean;
}

export interface UpdateFeatureRequest {
  id: number;
  name: string;
  description: string;
  isActive: boolean;
}

export interface FeatureFormData {
  name: string;
  description: string;
  isActive: boolean;
}
